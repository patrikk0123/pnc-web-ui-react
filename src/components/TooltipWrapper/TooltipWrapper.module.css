.tooltip-wrapper__tooltip a {
  color: var(--pf-t--global--text--color--inverse) !important;
}

.tooltip-wrapper__tooltip a:hover {
  color: var(--pf-t--global--color--nonstatus--gray--hover) !important;
}
