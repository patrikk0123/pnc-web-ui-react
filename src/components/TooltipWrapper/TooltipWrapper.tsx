import { Tooltip } from '@patternfly/react-core';
import { InfoCircleIcon } from '@patternfly/react-icons';
import { ReactNode } from 'react';

interface ITooltipWrapperProps {
  tooltip?: ReactNode;
  isVisible?: boolean;
}
/**
 * Tooltip Wrapper to make any component (like icon) to become a tooltip component.
 *
 * @param tooltip - the tooltip to be displayed
 * @param isVisible - set the tooltip visibility programmatically
 * @param children - the tooltip handler, by default <InfoCircleIcon />
 */
export const TooltipWrapper = ({ tooltip, isVisible, children }: React.PropsWithChildren<ITooltipWrapperProps>) =>
  tooltip ? (
    <Tooltip content={tooltip} isVisible={isVisible}>
      <> {children ? <>{children}</> : <InfoCircleIcon />}</>
    </Tooltip>
  ) : (
    <> {children ? <>{children}</> : null}</>
  );
