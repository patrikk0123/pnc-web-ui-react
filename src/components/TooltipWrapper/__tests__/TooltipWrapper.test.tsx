import { render } from '@testing-library/react';

import { TooltipWrapper } from 'components/TooltipWrapper/TooltipWrapper';

test('renders TooltipText', () => {
  render(<TooltipWrapper tooltip="test tooltip"></TooltipWrapper>);
});

test('compare snapshot with previous record', () => {
  const tree = render(<TooltipWrapper tooltip="test tooltip"></TooltipWrapper>);
  expect(tree.container).toMatchSnapshot();
});
