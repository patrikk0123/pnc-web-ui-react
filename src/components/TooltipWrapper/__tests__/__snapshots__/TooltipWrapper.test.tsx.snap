// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compare snapshot with previous record 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: contents;"
      >
         
        <svg
          aria-hidden="true"
          class="pf-v5-svg"
          fill="currentColor"
          height="1em"
          role="img"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: contents;"
    >
       
      <svg
        aria-hidden="true"
        class="pf-v5-svg"
        fill="currentColor"
        height="1em"
        role="img"
        viewBox="0 0 512 512"
        width="1em"
      >
        <path
          d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
